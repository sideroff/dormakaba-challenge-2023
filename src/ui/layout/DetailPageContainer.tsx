import { PropsWithChildren } from 'react';
import Stack from '@mui/material/Stack';

export function DetailPageContainer({ children }: PropsWithChildren) {
  return <Stack spacing={2}>{children}</Stack>;
}
