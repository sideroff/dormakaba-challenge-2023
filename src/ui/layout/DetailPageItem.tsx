import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import { PropsWithChildren } from 'react';

interface DetailPageItemProps {
  label: string;
}

export function DetailPageItem({
  label,
  children,
}: PropsWithChildren<DetailPageItemProps>) {
  return (
    <Box>
      <Typography>
        <strong>{label}</strong>
      </Typography>
      {children}
    </Box>
  );
}
