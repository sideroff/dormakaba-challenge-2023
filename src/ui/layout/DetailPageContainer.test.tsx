import { render } from '@testing-library/react';
import { DetailPageContainer } from './DetailPageContainer';
import { DetailPageItem } from './DetailPageItem';

describe('DetailPageContainer', () => {
  it('should render correctly', () => {
    const { container } = render(
      <DetailPageContainer>
        <DetailPageItem label="test">
          <div>Hello World!</div>
        </DetailPageItem>
      </DetailPageContainer>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
