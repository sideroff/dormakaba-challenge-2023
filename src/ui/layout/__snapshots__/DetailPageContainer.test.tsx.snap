// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPageContainer should render correctly 1`] = `
<div
  class="MuiStack-root css-pwj874-MuiStack-root"
>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <strong>
        test
      </strong>
    </p>
    <div>
      Hello World!
    </div>
  </div>
</div>
`;
